discord_slash/client.py

```diff
@@ -410,9 +411,30 @@ async def sync_all_commands(
                 self.logger.debug(
                     f"Detected changes on {scope if scope is not None else 'global'}, updating them"
                 )
-                existing_cmds = await self.req.put_slash_commands(
-                    slash_commands=to_send, guild_id=scope
-                )
+                try:
+                    existing_cmds = await self.req.put_slash_commands(
+                        slash_commands=to_send, guild_id=scope
+                    )
+                except discord.HTTPException as ex:
+                    if ex.status == 400:
+                        # catch bad requests
+                        cmd_nums = set(
+                            re.findall(r"In\s(\d).", ex.args[0])
+                        )  # find all discords references to commands
+                        error_string = ex.args[0]
+
+                        for num in cmd_nums:
+                            error_command = to_send[int(num)]
+                            error_string = error_string.replace(
+                                f"In {num}",
+                                f"'{error_command.get('name')}'",
+                            )
+
+                        ex.args = (error_string,)
+                        raise ex
+
+                    else:
+                        raise ex
```

Comment:
```suggestion

                    raise ex
```

Going to raise an exception regardless, the else part seems a bit redundant?

Reply:
Very true. I'll remove rectify that